### PR TITLE
fix: allow custom admin roles to be passed into gql transformer

### PR DESCRIPTION
**Pull request by SwaySway**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-provider-awscloudformation/src/graphql-transformer/utils.ts [hunk 1]
```diff
@@ -39,6 +44,7 @@ export const getIdentityPoolId = async (ctx: $TSContext): Promise<string | undef
 
 export const getAdminRoles = async (ctx: $TSContext, apiResourceName: string): Promise<Array<string>> => {
   const currentEnv = ctx.amplify.getEnvInfo().envName;
+  const backEndDir = ctx.amplify.pathManager.getBackendDirPath();
```
Comment (jhockett) [inline packages/amplify-provider-awscloudformation/src/graphql-transformer/utils.ts:47]: nit: use pathManager
```suggestion
  const backEndDir = pathManager.getBackendDirPath();
```

File: packages/amplify-provider-awscloudformation/src/graphql-transformer/utils.ts [hunk 2]
```diff
@@ -58,6 +64,15 @@ export const getAdminRoles = async (ctx: $TSContext, apiResourceName: string): P
     .filter((r: any) => r.dependsOn?.some((d: any) => d?.resourceName === apiResourceName))
     .map((r: any) => `${r.resourceName}-${currentEnv}`);
   adminRoles.push(...resources);
+
+  // check for custom iam admin roles
+  const customRoleFile = path.normalize(path.join(backEndDir, 'api', apiResourceName, CUSTOM_ROLES_FILE_NAME));
```
Comment (jhockett) [inline packages/amplify-provider-awscloudformation/src/graphql-transformer/utils.ts:69]:
```suggestion
  const customRoleFile = path.join(pathManager.getResourceDirectoryPath(undefined, AmplifyCategories.API, apiResourceName), CUSTOM_ROLES_FILE_NAME);
```

Comment (sachscode) [inline packages/amplify-provider-awscloudformation/src/graphql-transformer/utils.ts:69]: nit: can we use AmplifyCategories.API instead of api

**Review by jhockett — APPROVED** [review]
2 nits, but LGTM. amplify-cli-core import would need to be updated if accepting proposed suggestions.

**Review by sachscode — COMMENTED** [review]
LGTM, just one nit

**Review by jhockett — APPROVED** [review]

**Pull request merged**